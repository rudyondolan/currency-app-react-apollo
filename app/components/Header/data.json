
[
  { "key": "AUD", "text": "AUD", "value": "AUD" },
  { "key": "BGN", "text": "BGN", "value": "BGN" },
  { "key": "BRL", "text": "BRL", "value": "BRL" },
  { "key": "CAD", "text": "CAD", "value": "CAD" },
  { "key": "CHF", "text": "CHF", "value": "CHF" },
  { "key": "CNY", "text": "CNY", "value": "CNY" },
  { "key": "CZK", "text": "CZK", "value": "CZK" },
  { "key": "DKK", "text": "DKK", "value": "DKK" },
  { "key": "GBP", "text": "GBP", "value": "GBP" },
  { "key": "HKD", "text": "HKD", "value": "HKD" },
  { "key": "HRK", "text": "HRK", "value": "HRK" },
  { "key": "HUF", "text": "HUF", "value": "HUF" },
  { "key": "IDR", "text": "IDR", "value": "IDR" },
  { "key": "ILS", "text": "ILS", "value": "ILS" },
  { "key": "INR", "text": "INR", "value": "INR" },
  { "key": "ISK", "text": "ISK", "value": "ISK" },
  { "key": "JPY", "text": "JPY", "value": "JPY" },
  { "key": "KRW", "text": "KRW", "value": "KRW" },
  { "key": "MXN", "text": "MXN", "value": "MXN" },
  { "key": "MYR", "text": "MYR", "value": "MYR" },
  { "key": "NOK", "text": "NOK", "value": "NOK" },
  { "key": "NZD", "text": "NZD", "value": "NZD" },
  { "key": "PHP", "text": "PHP", "value": "PHP" },
  { "key": "PLN", "text": "PLN", "value": "PLN" },
  { "key": "RON", "text": "RON", "value": "RON" },
  { "key": "RUB", "text": "RUB", "value": "RUB" },
  { "key": "SEK", "text": "SEK", "value": "SEK" },
  { "key": "SGD", "text": "SGD", "value": "SGD" },
  { "key": "THB", "text": "THB", "value": "THB" },
  { "key": "TRY", "text": "TRY", "value": "TRY" },
  { "key": "USD", "text": "USD", "value": "USD" },
  { "key": "ZAR", "text": "ZAR", "value": "ZAR" }
]
