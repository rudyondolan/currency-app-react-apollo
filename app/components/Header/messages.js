/*
 * Header Messages
 *
 * This contains all the text for the Header component.
 */

import { defineMessages } from 'react-intl';

export default defineMessages({
  list: {
    id: 'boilerplate.components.Header.list',
    defaultMessage: 'List Currency',
  },
  about: {
    id: 'boilerplate.components.Header.about',
    defaultMessage: 'About',
  },
});
