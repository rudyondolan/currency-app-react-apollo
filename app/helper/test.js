import React from 'react';
import { IntlProvider, intlShape } from 'react-intl';
import { mount, shallow } from 'enzyme';

const intlProvider = new IntlProvider({ locale: 'en' }, {});
const { intl } = intlProvider.getChildContext();

function nodeWithIntlProp(node) {
  return React.cloneElement(node, { intl });
}

export function shallowWithIntl(node, { context, ...additionalOptions } = {}) {
  return shallow(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    ...additionalOptions,
  });
}

export function mountWithIntl(
  node,
  { context, childContextTypes, ...additionalOptions } = {},
) {
  return mount(nodeWithIntlProp(node), {
    context: Object.assign({}, context, { intl }),
    childContextTypes: Object.assign(
      {},
      { intl: intlShape },
      childContextTypes,
    ),
    ...additionalOptions,
  });
}
