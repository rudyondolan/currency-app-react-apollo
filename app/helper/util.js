export function toFloat(value) {
  if (typeof value === 'string') {
    const fVal = parseFloat(value);
    if (typeof fVal === 'number' && !isNaN(fVal)) {
      return fVal;
    }

    return 1;
  }

  return value;
}

