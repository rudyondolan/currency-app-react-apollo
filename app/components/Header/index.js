import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Menu, Container, Image, Input, Select } from 'semantic-ui-react';
import { FormattedMessage } from 'react-intl';
import Logo from 'assets/logo.png';
import messages from './messages';

const options = [
  { key: 'AUD', text: 'AUD', value: 'AUD' },
  { key: 'BGN', text: 'BGN', value: 'BGN' },
  { key: 'BRL', text: 'BRL', value: 'BRL' },
  { key: 'CAD', text: 'CAD', value: 'CAD' },
  { key: 'CHF', text: 'CHF', value: 'CHF' },
  { key: 'CNY', text: 'CNY', value: 'CNY' },
  { key: 'CZK', text: 'CZK', value: 'CZK' },
  { key: 'DKK', text: 'DKK', value: 'DKK' },
  { key: 'GBP', text: 'GBP', value: 'GBP' },
  { key: 'HKD', text: 'HKD', value: 'HKD' },
  { key: 'HRK', text: 'HRK', value: 'HRK' },
  { key: 'HUF', text: 'HUF', value: 'HUF' },
  { key: 'IDR', text: 'IDR', value: 'IDR' },
  { key: 'ILS', text: 'ILS', value: 'ILS' },
  { key: 'INR', text: 'INR', value: 'INR' },
  { key: 'ISK', text: 'ISK', value: 'ISK' },
  { key: 'JPY', text: 'JPY', value: 'JPY' },
  { key: 'KRW', text: 'KRW', value: 'KRW' },
  { key: 'MXN', text: 'MXN', value: 'MXN' },
  { key: 'MYR', text: 'MYR', value: 'MYR' },
  { key: 'NOK', text: 'NOK', value: 'NOK' },
  { key: 'NZD', text: 'NZD', value: 'NZD' },
  { key: 'PHP', text: 'PHP', value: 'PHP' },
  { key: 'PLN', text: 'PLN', value: 'PLN' },
  { key: 'RON', text: 'RON', value: 'RON' },
  { key: 'RUB', text: 'RUB', value: 'RUB' },
  { key: 'SEK', text: 'SEK', value: 'SEK' },
  { key: 'SGD', text: 'SGD', value: 'SGD' },
  { key: 'THB', text: 'THB', value: 'THB' },
  { key: 'TRY', text: 'TRY', value: 'TRY' },
  { key: 'USD', text: 'USD', value: 'USD' },
  { key: 'ZAR', text: 'ZAR', value: 'ZAR' },
];

class Header extends React.PureComponent {
  constructor(props) {
    super(props);
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handleSelectChange = this.handleSelectChange.bind(this);
  }

  handleInputChange(e) {
    const { onSetAmount } = this.props;
    const { value } = e.target;
    e.preventDefault();
    onSetAmount(value);
  }

  handleSelectChange(e, data) {
    const { onSetBase } = this.props;
    e.preventDefault();
    onSetBase(data.value);
  }

  render() {
    const { base, amount } = this.props;

    /* eslint-disable jsx-a11y/anchor-is-valid */
    return (
      <Menu fixed="top">
        <Container>
          <Menu.Item as="a" header>
            <Image size="mini" src={Logo} style={{ marginRight: '1.5em' }} />
            Fereign Exchange Currency
          </Menu.Item>
          <Link className="item" to="/">
            <FormattedMessage {...messages.list} />
          </Link>
          <Link className="item" to="/about">
            <FormattedMessage {...messages.about} />
          </Link>
          <Menu.Item position="right">
            <Input type="text" placeholder="Amount..." action>
              <input
                type="number"
                value={amount}
                onChange={this.handleInputChange}
              />
              <Select
                compact
                options={options}
                defaultValue={base}
                onChange={this.handleSelectChange}
              />
            </Input>
          </Menu.Item>
        </Container>
      </Menu>
    );
  }
}

Header.propTypes = {
  base: PropTypes.string.isRequired,
  amount: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  onSetBase: PropTypes.func.isRequired,
  onSetAmount: PropTypes.func.isRequired,
};

export default Header;
