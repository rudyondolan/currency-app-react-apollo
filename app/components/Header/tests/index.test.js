import React from 'react';
import { shallowWithIntl } from 'helper/test';
import Header from '../index';

describe('<Header />', () => {
  test('should match the snapshot', () => {
    const wrapper = renderHeader();
    expect(wrapper).toMatchSnapshot();
  });

  // test('should call change event on input', () => {
  //   const onSetAmount = jest.fn();
  //   const props = { onSetAmount };
  //   const wrapper = renderHeader(props);
  //   wrapper.find('input').simulate('change');
  // });
});

function renderHeader(props = {}) {
  const propsToUse = {
    base: 'USD',
    amount: 1,
    onSetBase() {},
    onSetAmount() {},
    ...props,
  };

  return shallowWithIntl(<Header {...propsToUse} />);
}
