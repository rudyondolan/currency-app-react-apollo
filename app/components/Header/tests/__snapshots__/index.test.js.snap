// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match the snapshot 1`] = `
<Menu
  fixed="top"
>
  <Container>
    <MenuItem
      as="a"
      header={true}
    >
      <Image
        as="img"
        size="mini"
        src="IMAGE_MOCK"
        style={
          Object {
            "marginRight": "1.5em",
          }
        }
        ui={true}
      />
      Fereign Exchange Currency
    </MenuItem>
    <Link
      className="item"
      replace={false}
      to="/"
    >
      <FormattedMessage
        defaultMessage="List Currency"
        id="boilerplate.components.Header.list"
        values={Object {}}
      />
    </Link>
    <Link
      className="item"
      replace={false}
      to="/about"
    >
      <FormattedMessage
        defaultMessage="About"
        id="boilerplate.components.Header.about"
        values={Object {}}
      />
    </Link>
    <MenuItem
      position="right"
    >
      <Input
        action={true}
        placeholder="Amount..."
        type="text"
      >
        <input
          onChange={[Function]}
          type="number"
          value={1}
        />
        <Select
          compact={true}
          defaultValue="USD"
          onChange={[Function]}
          options={
            Array [
              Object {
                "key": "AUD",
                "text": "AUD",
                "value": "AUD",
              },
              Object {
                "key": "BGN",
                "text": "BGN",
                "value": "BGN",
              },
              Object {
                "key": "BRL",
                "text": "BRL",
                "value": "BRL",
              },
              Object {
                "key": "CAD",
                "text": "CAD",
                "value": "CAD",
              },
              Object {
                "key": "CHF",
                "text": "CHF",
                "value": "CHF",
              },
              Object {
                "key": "CNY",
                "text": "CNY",
                "value": "CNY",
              },
              Object {
                "key": "CZK",
                "text": "CZK",
                "value": "CZK",
              },
              Object {
                "key": "DKK",
                "text": "DKK",
                "value": "DKK",
              },
              Object {
                "key": "GBP",
                "text": "GBP",
                "value": "GBP",
              },
              Object {
                "key": "HKD",
                "text": "HKD",
                "value": "HKD",
              },
              Object {
                "key": "HRK",
                "text": "HRK",
                "value": "HRK",
              },
              Object {
                "key": "HUF",
                "text": "HUF",
                "value": "HUF",
              },
              Object {
                "key": "IDR",
                "text": "IDR",
                "value": "IDR",
              },
              Object {
                "key": "ILS",
                "text": "ILS",
                "value": "ILS",
              },
              Object {
                "key": "INR",
                "text": "INR",
                "value": "INR",
              },
              Object {
                "key": "ISK",
                "text": "ISK",
                "value": "ISK",
              },
              Object {
                "key": "JPY",
                "text": "JPY",
                "value": "JPY",
              },
              Object {
                "key": "KRW",
                "text": "KRW",
                "value": "KRW",
              },
              Object {
                "key": "MXN",
                "text": "MXN",
                "value": "MXN",
              },
              Object {
                "key": "MYR",
                "text": "MYR",
                "value": "MYR",
              },
              Object {
                "key": "NOK",
                "text": "NOK",
                "value": "NOK",
              },
              Object {
                "key": "NZD",
                "text": "NZD",
                "value": "NZD",
              },
              Object {
                "key": "PHP",
                "text": "PHP",
                "value": "PHP",
              },
              Object {
                "key": "PLN",
                "text": "PLN",
                "value": "PLN",
              },
              Object {
                "key": "RON",
                "text": "RON",
                "value": "RON",
              },
              Object {
                "key": "RUB",
                "text": "RUB",
                "value": "RUB",
              },
              Object {
                "key": "SEK",
                "text": "SEK",
                "value": "SEK",
              },
              Object {
                "key": "SGD",
                "text": "SGD",
                "value": "SGD",
              },
              Object {
                "key": "THB",
                "text": "THB",
                "value": "THB",
              },
              Object {
                "key": "TRY",
                "text": "TRY",
                "value": "TRY",
              },
              Object {
                "key": "USD",
                "text": "USD",
                "value": "USD",
              },
              Object {
                "key": "ZAR",
                "text": "ZAR",
                "value": "ZAR",
              },
            ]
          }
        />
      </Input>
    </MenuItem>
  </Container>
</Menu>
`;
